src/include/platform/CHIPDeviceEvent.h

```diff
@@ -251,6 +251,16 @@ enum PublicEventTypes
      * sending messages to other nodes.
      */
     kServerReady,
+
+    /**
+     * Signals the application that device is going to factory reset.
+     */
+    kFactoryReset,
+
+    /**
```

Comment:
It would be good to have plumbing (including unit-tested plumbing) in FabricTable to handle removal of last fabric, which could then, separately, emit this event from Server::Server.

Reply:
Added way to emit event from Server